dna/keyRegistration/keyRegistration.js

```diff
@@ -1,35 +1,92 @@
   debug("Creating the user's keyRegistration");
-  //debug("choice :"+choice);
-  keyRegistration={perm_dpki_id:App.Agent.Hash,public_key:App.Key.Hash,shared_ID:App.Agent.String,revocation_Method_ID:choice};
-  //debug("keyRegistration="+keyRegistration);
-  directory=getDirectory();
+  revocation_Method_ID=selectRevocationMethord();
+  keyRegistration={perm_dpki_id:App.Agent.TopHash,public_key:App.Key.Hash,shared_ID:App.Agent.String,revocation_Method_ID:revocation_Method_ID};
+  me=getMeAgent();
   key=commit("keyRegistration",keyRegistration);
-  commit("user_keyRegistration_link", {Links:[{Base:directory,Link:key,Tag:"keyRegistration"}]});
-  debug("user_keyRegistration_link: "+JSON.stringify(getLink(directory,"keyRegistration",{Load:true})));
+  commit("user_keyRegistration_link", {Links:[{Base:me,Link:key,Tag:"keyRegistration"}]});
+  debug("user_keyRegistration_link: "+JSON.stringify(getLink(me,"keyRegistration",{Load:true})));
+  a=getLink(me,"keyRegistration",{Load:true});
   debug("User keyRegistration Created with the preset revocation Methord to 1");
   debug("===========================Phase 1.1 End===========================");
+  debug("===========================Phase 1.2 Starting===========================");
+//Revocation Key can be used for further work from here
+  debug("revocationKey is ="+ makeHash(keyRegistration));
+  debug("===========================Phase 1.2 End===========================");
+//return for the testing the function
+return a.Links[0].H;
 }
-
-function selectRevocationMethord(choice){
-  // This is a multiple choice
+function selectRevocationMethord(){
+  // This is a multiple choice in the UI. return the choice that the  user makes
   // 1 = Revocation Key
   // 2 = M of N Revocation
   // 3 = Revocation Athority
+  choice="1";
   return choice;
 }
 
-function keyRegistrationRead(){
-return true;
+function keyRegistrationUpdate(revoked_key){
+  debug("++++Update key Registration+++++")
+  //me=revoked_key;
+  var kr = doGetLink(revoked_key,"keyRegistration");
+  var n = kr.length - 1;
+  debug("N="+n);
+  if (n >= 0) {
+  var oldKey = kr[n];
+  debug("oldkeyRegistration"+ JSON.stringify(oldKey))
+  revocation_Method_ID=selectRevocationMethord();
+
+  //TODO change the "2" when the revocation methord is called from the UI Hash actually changes
+  /*Done because the same vause is not replaced in the DHT wheich gives an ERROR*/
+  //new_keyRegistration={perm_dpki_id:App.Agent.Hash,public_key:App.Key.Hash,shared_ID:App.Agent.String,revocation_Method_ID:"2"};
+    new_keyRegistration={perm_dpki_id:App.Agent.TopHash,public_key:App.Key.Hash,shared_ID:App.Agent.String,revocation_Method_ID:revocation_Method_ID};
+  /*  debug("App.Agent.Hash="+App.Agent.Hash)
+    debug("App.AgentTop.Hash="+App.Agent.TopHash)
+    debug("App.Key.Hash="+App.Key.Hash)
+  */
+  var key = update("keyRegistration",new_keyRegistration,oldKey);
+  debug(new_keyRegistration+" is "+key);
+  commit("user_keyRegistration_link",
```

Comment:
I'm not sure I full understand this, but it seems to me that this commit of links should include a link for the  new key link, or does that happen somewhere else?

I don't see tests for these functions.  I would like to see some tests where after calling keyRegistrationUpdate()  there's a call to to some functions that make it clear what should happen, i.e. what should now be available on the DHT if you do a doGetLink() call.